Odata-docs/webapi-8/fundamentals/server-driven-paging.md

```diff
@@ -0,0 +1,251 @@
+**Response:**
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Products",
+    "value": [
+        {
+            "Id": 3,
+            "Name": "Product 3"
+        }
+    ]
+}
+```
+This is now the last page. The server returns only one item and no next link.
+
+## Improving paging with `$skiptoken`
+
+You may have noticed that ASP.NET Core OData generates next links using `$skip`. Using `$skip` on large collections may lead to performance degradation in some databases. For more information about the drawbaks of `$skip`, visit the [client-driven paging](/odata/webapi-8/fundamentals/client-driven-paging#drawbacks) article.
+
+OData provides a [`$skiptoken`](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_ServerDrivenPaging) query option that can be used in server-driven paging to encode information about the next page of a result. The content of `$skiptoken` is opaque and service-specific. The client should not try to interpret the value or rely on its format. OData clients must not use `$skiptoken` when constructing requests.
+
+ASP.NET Core OData 8 provides support for pagination based on `$skiptoken`, but it's not enabled by default. You can enable it using the [`ODataOptions.SkipToken()`](/dotnet/api/microsoft.aspnetcore.odata.odataoptions.skiptoken) method when configuring OData services in your application:
+
+```c#
+services.AddControllers().AddOData(options =>
+    options.SetMaxTop(null).SkipToken());
+```
+
+Assuming we have the same products entity set as in the previous section and a `PageSize` of 2, here's what the first page would look like:
+
+```http
+GET http://localhost:5000/Products
+```
+
+**Response**
+
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Products",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Product 1"
+        },
+        {
+            "Id": 2,
+            "Name": "Product 2"
+        }
+    ],
+    "@odata.nextLink": "http://localhost:64771/Products?$skiptoken=Id-2"
+}
+```
+The next link is now generated based on `$skiptoken` rather than `$skip`. This is what the second page would look like:
+
+```
+GET http://localhost:5000/Products?$skiptoken=Id-2
+```
+
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Products",
+    "value": [
+        {
+            "Id": 3,
+            "Name": "Product 3"
+        },
+        {
+            "Id": 4,
+            "Name": "Product 4"
+        }
+    ],
+    "@odata.nextLink": "http://localhost:64771/Products?$skiptoken=Id-4"
+}
+```
+
+The `$skiptoken` value is generated using an implementation of [`SkipTokenHandler`](/dotnet/api/microsoft.aspnetcore.odata.query.skiptokenhandler). By default, the built-in [`DefaultSkipTokenHandler`](/dotnet/api/microsoft.aspnetcore.odata.query.defaultskiptokenhandler) class. `DefaultSkipTokenHandler` generates the `$skiptoken` based on the values of the key fields and fields in the `$orderby` query option if present.
```

Comment:
```suggestion
The `$skiptoken` value is generated using an implementation of [`SkipTokenHandler`](/dotnet/api/microsoft.aspnetcore.odata.query.skiptokenhandler), by default the built-in [`DefaultSkipTokenHandler`](/dotnet/api/microsoft.aspnetcore.odata.query.defaultskiptokenhandler) class. `DefaultSkipTokenHandler` generates the `$skiptoken` based on the values of the key fields and fields in the `$orderby` query option if present.
```

Reply:
Would that not make the sentence too long?

Reply:
@habbes It's because "By default, the built-in DefaultSkipTokenHandler class." doesn't feel like a complete sentence

Reply:
By default, the built-in [`DefaultSkipTokenHandler`](/dotnet/api/microsoft.aspnetcore.odata.query.defaultskiptokenhandler) class generates the `$skiptoken` based on the values of the key fields and fields in the `$orderby` query option if present

What about this?

Reply:
Aah I see, my bad.

Reply:
Rephrased to "By default, the built-in `DefaultSkipTokenHandler` class generates..."